Refs #25588 -- Fixed GDAL dependency in spatial lookups.

yellowcap opened the pull request:
I moved the attempt to convert to raster into a separate method that is only called if GDAL is installed. 

Is there a way to create a regression test for this? I don't know how to mock a system that does not have gdal installed.

timgraham commented:
I don't think any tests are needed. This works besides the test failure reported by Jenkins and a few other issues, at least some of which don't appear to be related to this ticket:
```
======================================================================
ERROR: test_srid_option (gis_tests.geoapp.test_serializers.GeoJSONSerializerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/media/sf_django/tests/gis_tests/geoapp/test_serializers.py", line 85, in test_srid_option
    geojson = serializers.serialize('geojson', City.objects.all().order_by('name'), srid=2847)
  File "/media/sf_django/django/core/serializers/__init__.py", line 129, in serialize
    s.serialize(queryset, **options)
  File "/media/sf_django/django/core/serializers/base.py", line 96, in serialize
    self.end_object(obj)
  File "/media/sf_django/django/core/serializers/json.py", line 61, in end_object
    json.dump(self.get_dump_object(obj), self.stream,
  File "/media/sf_django/django/contrib/gis/serializers/geojson.py", line 58, in get_dump_object
    'Unable to convert geometry to SRID %s when GDAL is not installed.' % self.srid
django.core.serializers.base.SerializationError: Unable to convert geometry to SRID 2847 when GDAL is not installed.

======================================================================
ERROR: test06_f_expressions (gis_tests.relatedapp.tests.RelatedGeoModelTest)
Testing F() expressions on GeometryFields.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/media/sf_django/tests/gis_tests/relatedapp/tests.py", line 155, in test06_f_expressions
    c2 = c1.transform(2276, clone=True)
  File "/media/sf_django/django/contrib/gis/geos/geometry.py", line 524, in transform
    raise GEOSException("GDAL library is not available to transform() geometry.")
django.contrib.gis.geos.error.GEOSException: GDAL library is not available to transform() geometry.

======================================================================
FAIL: test_add_raster_field_on_db_without_raster_support (gis_tests.gis_migrations.test_operations.OperationTests)
----------------------------------------------------------------------
django.core.exceptions.ImproperlyConfigured: RasterField requires GDAL.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/media/sf_django/django/test/testcases.py", line 1092, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/media/sf_django/tests/gis_tests/gis_migrations/test_operations.py", line 134, in test_add_raster_field_on_db_without_raster_support
    False, fields.RasterField
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/media/sf_django/django/test/testcases.py", line 604, in _assert_raises_message_cm
    self.assertIn(expected_message, str(cm.exception))
AssertionError: 'Raster fields require backends with raster support.' not found in 'RasterField requires GDAL.'

======================================================================
FAIL: test_create_raster_model_on_db_without_raster_support (gis_tests.gis_migrations.test_operations.OperationTests)
----------------------------------------------------------------------
django.core.exceptions.ImproperlyConfigured: RasterField requires GDAL.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/media/sf_django/django/test/testcases.py", line 1092, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/media/sf_django/tests/gis_tests/gis_migrations/test_operations.py", line 123, in test_create_raster_model_on_db_without_raster_support
    self.set_up_test_model(True)
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/media/sf_django/django/test/testcases.py", line 604, in _assert_raises_message_cm
    self.assertIn(expected_message, str(cm.exception))
AssertionError: 'Raster fields require backends with raster support.' not found in 'RasterField requires GDAL.'
```

yellowcap commented:
I fixed the test failures from Jenkins, and added `@skipUnless(HAS_GDAL)` decorators to the two raster support db tests. I suspect these would have failed on a system with PostGIS and without GDAL even before adding the lookups.

The other two tests do not seem to be related to the ticket.
